src/Integration/RequestIntegration.php

```diff
@@ -55,24 +72,38 @@ public static function applyToEvent(Event $event, ?ServerRequestInterface $reque
         $requestData = [
             'url' => (string) $request->getUri(),
             'method' => $request->getMethod(),
-            'headers' => $request->getHeaders(),
-            'cookies' => $request->getCookieParams(),
         ];
 
         if ($request->getUri()->getQuery()) {
             $requestData['query_string'] = $request->getUri()->getQuery();
         }
 
-        if ($request->hasHeader('REMOTE_ADDR')) {
-            $requestData['env']['REMOTE_ADDR'] = $request->getHeaderLine('REMOTE_ADDR');
+        if ($self->options->shouldSendDefaultPii()) {
+            if ($request->hasHeader('REMOTE_ADDR')) {
+                $requestData['env']['REMOTE_ADDR'] = $request->getHeaderLine('REMOTE_ADDR');
+            }
+
+            $requestData['cookies'] = $request->getCookieParams();
+            $requestData['headers'] = $request->getHeaders();
+
+            $userContext = $event->getUserContext();
+
+            if (null === $userContext->getIpAddress() && $request->hasHeader('REMOTE_ADDR')) {
+                $userContext->setIpAddress($request->getHeaderLine('REMOTE_ADDR'));
+            }
+        } else {
+            $requestData['headers'] = $self->removePiiFromHeaders($request->getHeaders());
         }
 
         $event->setRequest($requestData);
+    }
 
-        $userContext = $event->getUserContext();
+    private function removePiiFromHeaders(array $headers): array
```

Comment:
This function is missing the docblock (description of both the method and the parameter)